vms/platformvm/state/state.go

```diff
@@ -2132,6 +2379,65 @@ func (s *state) updateValidatorManager(updateValidators bool) error {
 		}
 	}
 
+	// Remove all deleted SoV validators. This must be done before adding new
+	// SoV validators to support the case where a validator is removed and then
+	// immediately re-added with a different validationID.
+	for validationID, sov := range s.sovDiff.modified {
+		if !sov.isDeleted() {
+			continue
+		}
+
+		priorSOV, err := s.getPersistedSubnetOnlyValidator(validationID)
+		if err == database.ErrNotFound {
+			// Deleting a non-existent validator is a noop. This can happen if
+			// the validator was added and then immediately removed.
+			continue
+		}
+		if err != nil {
+			return err
+		}
+
+		if err := s.validators.RemoveWeight(priorSOV.SubnetID, priorSOV.effectiveNodeID(), priorSOV.Weight); err != nil {
+			return err
+		}
+	}
+
+	// Now that the removed SoV validators have been deleted, perform additions
+	// and modifications.
+	for validationID, sov := range s.sovDiff.modified {
+		if sov.isDeleted() {
+			continue
+		}
+
+		priorSOV, err := s.getPersistedSubnetOnlyValidator(validationID)
+		switch err {
+		case nil:
+			// Modifying an existing validator
+			if priorSOV.isActive() == sov.isActive() {
+				// This validator's active status isn't changing. This means
+				// the effectiveNodeIDs are equal.
+				nodeID := sov.effectiveNodeID()
+				if priorSOV.Weight < sov.Weight {
+					err = s.validators.AddWeight(sov.SubnetID, nodeID, sov.Weight-priorSOV.Weight)
+				} else if priorSOV.Weight > sov.Weight {
+					err = s.validators.RemoveWeight(sov.SubnetID, nodeID, priorSOV.Weight-sov.Weight)
```

Comment:
is `priorSOV.Weight > sov.Weight` legal? what does it mean?

Reply:
If the `priorSOV.Weight` is greater than `sov.Weight`, then the weight of the validator is being decreased by `priorSOV.Weight - sov.Weight` units

Reply:
no but i asked what it means when they're equal, I edited shortly after, probably github didn't refresh my comment

Reply:
Ah. It is possible for them to be equal. In that case the weight does not need to be modified.